U/danielsf/parallel/inst/cat

danielsf opened the pull request:
This should allow us to do parallelized InstanceCatalog generation

danielsf committed:
give generateInstCat.py a __main__

danielsf committed:
use makedirs instead of mkdir

danielsf committed:
globally import InstanceCatalogWriter
(automatically suppress warnings for that step)

danielsf committed:
try to parallelize instance catalog generation

danielsf committed:
fix bug

danielsf committed:
fix typo

danielsf committed:
fix other bug

danielsf committed:
write to correct joblog

danielsf committed:
print statement

danielsf committed:
another print statement

danielsf committed:
even more print statements

danielsf committed:
fix iteration of obsHistID

danielsf committed:
just use kwargs for generation method

danielsf committed:
put newlines at end of log print statements

danielsf committed:
add test_script.sl

danielsf committed:
move assignment of catalog members to before write_catalog

danielsf committed:
write_catalog can write out status for obsHistID

danielsf committed:
use status_dir

danielsf committed:
remove print statements

danielsf committed:
up the time on test_script.sl

danielsf committed:
back to 2 cpus per task

danielsf committed:
update test script to only call srun once

danielsf committed:
set -n == -N to get cores allotted correctly

danielsf committed:
this test script configuration seems to work

danielsf committed:
print statements

jchiang87 commented on bin.src/generateInstCat.py:
```diff
@@ -2,73 +2,147 @@
+                                                   protoDC2_dec=args.protoDC2_dec,
+                                                   agn_db_name=args.agn_db_name,
+                                                   sn_db_name=args.sn_db_name,
+					           host_image_dir=args.host_image_dir,
```
Can we fix the indentation here? (I meant to ask about his in the lensing PR, but forgot.)

danielsf replied:
Looks like it was a stray tab.

jchiang87 reviewed (approved):
Looks good Scott!  Thanks.

danielsf committed:
back to many sruns

danielsf committed:
write which node it is using

danielsf committed:
get actual host name from /etc/hostname

danielsf committed:
-n 2 for each srun

danielsf committed:
fix indent

danielsf committed:
put obsHistID into status log lines

danielsf committed:
back to -n 1

when -n was 2, I got a weird warning from slurm about
no longer being able to enforce --tasks-per-node=1.  The log files
also contained double entries, though I'm not sure that had any
consequences for the output instanceCatalogs
